junit-runtime/src/main/scala/org/hamcrest/BaseDescription.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Ported from https://github.com/hamcrest/JavaHamcrest/
+ */
+package org.hamcrest
+
+import java.util.Arrays
+import scala.annotation.tailrec
+import org.hamcrest.internal.ArrayIterator
+import org.hamcrest.internal.SelfDescribingValueIterator
+
+abstract class BaseDescription extends Description {
+  override def appendText(text: String): Description = {
+    append(text)
+    this
+  }
+
+  override def appendDescriptionOf(value: SelfDescribing): Description = {
+    value.describeTo(this)
+    this
+  }
+
+  override def appendValue(value: AnyRef): Description = {
+    value match {
+      case null                   => append("null")
+      case v: String              => append(toJavaSyntax(v))
+      case v: java.lang.Character =>
```

Comment:
Blank line above this line, and then put the arrows of the first two line closer to `: String`. Or simply expand everything to have a uniform style:
```scala
case null =>
  append("null")

case v: String =>
  append(toJavaSyntax(v))

case v: java.lang.Character => // could be Char, btw
  ...
```
Also, we typically reuse the identifier when refining types in pattern matching, so use `value` instead of `v`.